### DISCUSSION TITLE
Troubleshoot Spacing with Piping on Windows

### DISCUSSION THREAD

**Discussion by hay-kot** [JanDeDobbeleer/oh-my-posh#2607] [Troubleshoot] [answered]
Hey! First off, I don't think this is an issue with Oh-My-Posh my guess is that it has to do with some rendering in the underlying Go library I'm using or a Window Terminal issue. 

Related Issues
- https://github.com/charmbracelet/gum/issues/32 - He goes into a bit more detail on the several underlying issues, but also provides a good example of how the prompt get's messed up after running the command.

I have a program that I'm using to fuzzy search through a list of project and then pass that to Set-Location to change directories into that. There were some other errors that I had to resolve to get it to work since it was printing errors to the screen because of the underlying Go Library I'm using. 

**My actual problem** that I can't solve is that it mucks up my Oh-My-Posh themes spacing and I can't seem to get it to reset. I've tried

- rerunning `Set-PoshPrompt`,
- clearing the terminal, 
- doing a full profile reload 

but the spacing issue continues until I open a new tab. **Does anyone have any clever ideas for a work around so it doesn't mangle my prompt?** 

**Example of Error**

![Screenshot 2022-08-02 123753](https://user-images.githubusercontent.com/64056131/182470286-7b99145b-2231-473b-bb71-176d1b4b69fc.png)

**Comment by JanDeDobbeleer** [comment]
First thing I would do is update oh-my-posh. You're still on the module and that's been gone for over 6 months. See [here](https://ohmyposh.dev/docs/migrating) for reference.

**Reply by hay-kot** [reply]
Yikes! Not sure how that slipped my mind. Thanks for the link.

Updating didn't make any difference on the issue unfortunately.

**Comment by JanDeDobbeleer** [comment, marked as answer]
@hay-kot had a read through everything and I'm pretty sure there's not much we can do on our end. Somehow gum messes with the ansi sequences or terminal interpretation which causes the offset.

**Reply by hay-kot** [reply]
Ah bummer.

Thanks for taking a look, and all the work you do with OMP!
